### PR TITLE
Refactor Brave Ads unit tests/mocks/utils

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/9471

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/purchase_intent_classifier_unittest.cc [hunk 1]
```diff
@@ -21,10 +21,13 @@
 
 using ::testing::_;
 
-// npm run test -- brave_unit_tests --filter=AdsPurchaseIntentClassifier*
+// npm run test -- brave_unit_tests --filter=TerryToDo*
```
Comment (emerick) [inline vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/purchase_intent_classifier_unittest.cc:24]: Intentional change?

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/purchase_intent_classifier_unittest.cc:24]: Ah yes, WIP. Just 4 files left to refactor

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/frequency_capping_unittest_utils.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#include "base/time/time.h"
+
+namespace ads {
+
+void GeneratePastCreativeSetHistoryFromNow(
+    Client* client,
+    const std::string& creative_set_id,
+    const int64_t time_offset_in_seconds,
+    const uint8_t count) {
+  uint64_t timestamp_in_seconds = base::Time::Now().ToDoubleT();
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/frequency_capping_unittest_utils.cc:22]: Fixed and will raise another issue to fix the rest of Brave Ads

File: vendor/bat-native-ads/src/bat/ads/internal/ads_unittest_utils.cc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::Return;
+
+namespace ads {
+
+base::FilePath GetTestDataPath() {
+  base::FilePath path;
+  base::PathService::Get(base::DIR_SOURCE_ROOT, &path);
+  path = path.AppendASCII("brave/vendor/bat-native-ads/data/test");
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_unittest_utils.cc:28]: Fixed

**Pull request closed without merging**